### ISSUE TITLE
Support Atlantis merge checkout strategy

### ISSUE THREAD

**Issue by jayceebernardino** [infracost/infracost#402]
We are using atlantis' [merge](https://www.runatlantis.io/docs/checkout-strategy.html#merge) strategy

Atlantis performs a local merge so we get 

```
Exiting as the current branch was the default branch so nothing more to do.
```
Because the git rev-parse output is master when the PR is trying to merge to master

ie:
```
git log

commit e72bd23e80ecd1f81af628022a716e46d3db5725 (HEAD -> master)
Merge: 64c7a1f853 060b27c9cc
Author: atlantis <atlantis@runatlantis.io>
Date:   Mon Feb 8 21:28:02 2021 +0000

    atlantis-merge
```

**Comment by alikhajeh1** [comment]
@jayceebernardino thanks for reporting this and #403, I think I've fixed them in https://github.com/infracost/infracost/pull/404 and created a temporary [Docker image](https://github.com/infracost/infracost-atlantis/blob/master/Dockerfile) with that updated script.

Can you possibly try running `infracost/infracost-atlantis:temp_test` instead of `infracost/infracost-atlantis:latest` and let me know if that works? I can then merge that PR and update `latest`

**Comment by jayceebernardino** [comment]
> @jayceebernardino thanks for reporting this and #403, I think I've fixed them in #404 and created a temporary [Docker image](https://github.com/infracost/infracost-atlantis/blob/master/Dockerfile) with that updated script.
> 
> Can you possibly try running `infracost/infracost-atlantis:temp_test` instead of `infracost/infracost-atlantis:latest` and let me know if that works? I can then merge that PR and update `latest`

Thanks you so much for your help on this. I made one modification to your change to make it work.  In https://github.com/infracost/infracost/blob/9c1af38751d08c6cba4d198a5605174aa6c1610e/scripts/ci/atlantis_diff.sh#L46-L50

I replaced `origin` with `head` (Probably because we create pull requests from forks internally )  

```
# In atlantis pod
bash-5.0$ git remote -v 
head <fork url>
origin <origin url>
```

**Comment by alikhajeh1** [comment]
Thanks for testing @jayceebernardino! I looked into the [Atlantis code](https://github.com/runatlantis/atlantis/blob/151308998829ebfd1cfd1fb020c27576a65fa339/server/events/working_dir.go#L141-L144) and they use "head" as the git remote for the head branch (base branch uses "origin") so I've [updated](https://github.com/infracost/infracost/commit/d825035affa6b3a205511019cbe6130c3d151a58#diff-d6ad9a31bd708d0c0b3b2bc79989cd7c7f7693f782295afb47a1cf156f95c52bR46-R50) the script to match what you did.

The Docker image `infracost/infracost-atlantis:latest` now has this fix.
